### PR TITLE
Simpler License Headers

### PR DESCRIPTION
**Pull request by trexfeathers**
Addresses #3285. 

In line with the latest [SciTools licensing guidelines](https://github.com/SciTools/scitools.org.uk/blob/master/documents/scitools_licensing.md#source-preamble), the license headers are much shorter and mostly reference the master file in the root directory.

Also in line with these guidelines, the COPYING.LESSER file has been refactored to be called LICENSE, while COPYING has been removed since it was not being referenced.

The years test in test_coding_standards.py has been replaced with a basic check that each file starts with the correct string.

### CODE REVIEW

**Comment by trexfeathers** [comment]
Apologies for the slightly messy commit sequence - was originally just aiming for changing the years

**Comment by trexfeathers** [comment]
Not sure how GitHub should be detecting the license, since LICENSE is _supposed_ to be the name and _COPYING.LESSER_ was an unconventional alternative

**Comment by QuLogic** [comment]
> Also in line with these guidelines, the COPYING.LESSER file has been refactored to be called LICENSE, while COPYING has been removed since it was not being referenced.

This is incorrect. The `COPYING.LESSER` are _additional_ permissions of the LGPLv3 wrt to the GPLv3. You still need the `COPYING` file. It is also just as old a name as `LICENSE`, so I would not call it unconventional.

**Comment by bjlittle** [comment]
@trexfeathers For some context, see [How to use GNU licenses for your own software](https://www.gnu.org/licenses/gpl-howto.en.html)...

That said, at the moment I'd suggest that `COPYING` is reinstated as was, along with `COPYING.LESSER` as was.

When, and if, we move to a `BSD 3-Clause New` license, then we can remove the `COPYING` and `COPYING.LESSER`, and then use the `LICENSE` file as it's sole replacement.

File: MANIFEST.in [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Top-level files
-include CHANGES COPYING COPYING.LESSER INSTALL
+include CHANGES LICENSE INSTALL
```
Comment (bjlittle) [inline MANIFEST.in]: This will need changed back...

File: README.md [hunk 2]
```diff
@@ -112,7 +112,7 @@ are available in [INSTALL](INSTALL).
 # Copyright and licence
 
 Iris may be freely distributed, modified and used commercially under the terms
-of its [GNU LGPLv3 license](COPYING.LESSER).
+of its [GNU LGPLv3 license](LICENSE).
```
Comment (bjlittle) [inline README.md]: As was...

File: docs/iris/example_tests/__init__.py [hunk 3]
```diff
@@ -1,19 +1,7 @@
-# (C) British Crown Copyright 2010 - 2015, Met Office
+# Copyright Iris Contributors
```
Comment (bjlittle) [inline docs/iris/example_tests/__init__.py]: `# Copyright Iris contributors` not `# Copyright Iris Contributors`

File: docs/iris/example_tests/__init__.py [hunk 4]
```diff
@@ -1,19 +1,7 @@
...
-#
-# Iris is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU Lesser General Public License for more details.
-#
-# You should have received a copy of the GNU Lesser General Public License
-# along with Iris.  If not, see <http://www.gnu.org/licenses/>.
+# This file is part of Iris and is released under the LGPL license.
+# See LICENSE in the root of the repository for full licensing details.
```
Comment (bjlittle) [inline docs/iris/example_tests/__init__.py]: This last line should then be...
```
# See COPYING and COPYING.LESSER in the root of the repository for
# full licensing details.
```

File: docs/iris/src/conf.py [hunk 5]
```diff
@@ -100,7 +88,7 @@
 # define the copyright information for latex builds. Note, for html builds,
 # the copyright exists directly inside "_templates/layout.html"
 upper_copy_year = datetime.datetime.now().year
-copyright = u'British Crown Copyright 2010 - {}, Met Office'.format(upper_copy_year)
+copyright = u'Copyright Iris Contributors'
```
Comment (bjlittle) [inline docs/iris/src/conf.py]: `Copyright Iris contributors`

File: docs/iris/src/copyright.rst [hunk 6]
```diff
@@ -2,34 +2,21 @@
...
-.. |copy|   unicode:: U+000A9 .. COPYRIGHT SIGN
-
 Iris code
 ---------
 
-All Iris source code, unless explicitly stated, is |copy| ``British Crown copyright, 2014`` and 
-is licensed under the **GNU Lesser General Public License** as published by the
-Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+All Iris source code, unless explicitly stated, is ``Copyright Iris
+Contributors`` and is licensed under the **GNU Lesser General Public
```
Comment (bjlittle) [inline docs/iris/src/copyright.rst]: `contributors`

File: docs/iris/src/copyright.rst [hunk 7]
```diff
@@ -2,34 +2,21 @@
...
-    Iris is free software: you can redistribute it and/or modify it under
-    the terms of the GNU Lesser General Public License as published by the
-    Free Software Foundation, either version 3 of the License, or
-    (at your option) any later version.
-    
-    Iris is distributed in the hope that it will be useful,
-    but WITHOUT ANY WARRANTY; without even the implied warranty of
-    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-    GNU Lesser General Public License for more details.
+    Copyright Iris Contributors
```
Comment (bjlittle) [inline docs/iris/src/copyright.rst]: `contributors`

File: docs/iris/src/copyright.rst [hunk 8]
```diff
@@ -2,34 +2,21 @@
...
     
-    You should have received a copy of the GNU Lesser General Public License
-    along with Iris.  If not, see `<http://www.gnu.org/licenses/>`_. 
+    This file is part of Iris and is released under the LGPL license.
+    See LICENSE in the root of the repository for full licensing details.
```
Comment (bjlittle) [inline docs/iris/src/copyright.rst]: `See COPYING and COPYING.LESSER in the root of the repository for full licensing details.`

File: lib/iris/tests/test_coding_standards.py [hunk 9]
```diff
@@ -35,30 +23,11 @@
...
-# (C) British Crown Copyright {YEARS}, Met Office
-#
-# This file is part of Iris.
-#
-# Iris is free software: you can redistribute it and/or modify it under
-# the terms of the GNU Lesser General Public License as published by the
-# Free Software Foundation, either version 3 of the License, or
-# (at your option) any later version.
+LICENSE_TEMPLATE = \
+    """# Copyright Iris Contributors
```
Comment (bjlittle) [inline lib/iris/tests/test_coding_standards.py]: `contributors`

File: lib/iris/tests/test_coding_standards.py [hunk 10]
```diff
@@ -35,30 +23,11 @@
...
-# along with Iris.  If not, see <http://www.gnu.org/licenses/>.""".strip()
-
-
-LICENSE_RE_PATTERN = re.escape(LICENSE_TEMPLATE).replace('\{YEARS\}', '(.*?)')
-# Add shebang possibility to the LICENSE_RE_PATTERN
-LICENSE_RE_PATTERN = r'(\#\!.*\n)?' + LICENSE_RE_PATTERN
-LICENSE_RE = re.compile(LICENSE_RE_PATTERN, re.MULTILINE)
-
+# This file is part of Iris and is released under the LGPL license.
+# See LICENSE in the root of the repository for full licensing details."""
```
Comment (bjlittle) [inline lib/iris/tests/test_coding_standards.py]: Change to match...

File: MANIFEST.in (hunk 1 above)
Comment (trexfeathers) [reply MANIFEST.in]: 👍

File: README.md (hunk 2 above)
Comment (trexfeathers) [reply README.md]: 👍

File: docs/iris/example_tests/__init__.py (hunk 3 above)
Comment (trexfeathers) [reply docs/iris/example_tests/__init__.py]: 👍

File: docs/iris/example_tests/__init__.py (hunk 4 above)
Comment (trexfeathers) [reply docs/iris/example_tests/__init__.py]: 👍

File: docs/iris/src/conf.py (hunk 5 above)
Comment (trexfeathers) [reply docs/iris/src/conf.py]: 👍

File: docs/iris/src/copyright.rst (hunk 6 above)
Comment (trexfeathers) [reply docs/iris/src/copyright.rst]: 👍

File: docs/iris/src/copyright.rst (hunk 7 above)
Comment (trexfeathers) [reply docs/iris/src/copyright.rst]: 👍

File: docs/iris/src/copyright.rst (hunk 8 above)
Comment (trexfeathers) [reply docs/iris/src/copyright.rst]: 👍

File: lib/iris/tests/test_coding_standards.py (hunk 9 above)
Comment (trexfeathers) [reply lib/iris/tests/test_coding_standards.py]: 👍

File: lib/iris/tests/test_coding_standards.py (hunk 10 above)
Comment (trexfeathers) [reply lib/iris/tests/test_coding_standards.py]: 👍

**Comment by trexfeathers** [comment]
Thank you @QuLogic and @bjlittle for the information on licensing. My latest changes include reverting what I had done with the license files

**Comment by bjlittle** [comment]
@trexfeathers Awesome, many thanks! 👍

**Pull request merged**